src/transformers/modeling_squeezebert.py

```diff
@@ -0,0 +1,1032 @@
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.activation = nn.Tanh()
+
+    def forward(self, hidden_states):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        first_token_tensor = hidden_states[:, 0]
+        pooled_output = self.dense(first_token_tensor)
+        pooled_output = self.activation(pooled_output)
+        return pooled_output
+
+
+class SqueezeBertPreTrainedModel(PreTrainedModel):
+    """An abstract class to handle weights initialization and
+    a simple interface for downloading and loading pretrained models.
+    """
+
+    config_class = SqueezeBertConfig
+    base_model_prefix = "transformer"
+    authorized_missing_keys = [r"position_ids"]
+
+    def _init_weights(self, module):
+        """ Initialize the weights """
+        if isinstance(module, (nn.Linear, nn.Conv1d, nn.Embedding)):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+        elif isinstance(module, SqueezeBertLayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        if isinstance(module, (nn.Linear, nn.Conv1d)) and module.bias is not None:
+            module.bias.data.zero_()
+
+
+SQUEEZEBERT_START_DOCSTRING = r"""    The SqueezeBERT model was proposed in
+    `SqueezeBERT: What can computer vision teach NLP about efficient neural networks?
+    <https://arxiv.org/abs/2006.11316>`__ by Forrest N. Iandola, Albert E. Shaw, Ravi Krishna, and Kurt W. Keutzer
+
+    This model is a PyTorch `torch.nn.Module <https://pytorch.org/docs/stable/nn.html#torch.nn.Module>`_ sub-class.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general
+    usage and behavior.
+
+    For best results finetuning SqueezeBERT on text classification tasks, it is recommended to use the
+    `squeezebert/squeezebert-mnli-headless` checkpoint as a starting point.
+
+    Parameters:
+        config (:class:`~transformers.SqueezeBertConfig`): Model configuration class with all the parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the configuration.
+            Check out the :meth:`~transformers.PreTrainedModel.from_pretrained` method to load the model weights.
+
+    Hierarchy::
+
+        Internal class hierarchy:
+            SqueezeBertModel
+                SqueezeBertEncoder
+                    SqueezeBertModule
+                    SqueezeBertSelfAttention
+                        CA  (convolution, activation)
+                        CDL (convolution, dropout, layernorm)
+
+    Data layouts::
+
+        Input data is in [N, W, C] format.
```

Comment:
Not sure why we use acronyms here with three lines below to explain what the letter means. It would be much easier to understand in a few sentences, and this should probably use the warning template to be better rendered in the docs:
```
.. warning::
    The input data and output of the encoder use the format :obj:`[batch_size, sequence_length, hidden_size]` but the data inside the encoder uses the :obj:`[batch_size, hidden_size, sequence_length]` format.
```

The :obj:` ` are there to code-format what's inside for the documentation.